src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -321,6 +321,13 @@ public override IntermediateNode Visit(FirstNameNode node, IRTranslatorContext c
                 Contracts.AssertValue(node);
                 Contracts.AssertValue(context);
 
+                var nodeType = context.Binding.GetType(node);
+
+                if (nodeType.IsDeferred)
+                {
+                    throw new NotSupportedException("Deferred(Unknown) is not supported");
```

Comment:
nit: this string is repeated 2 places, define a constant?
Maybe add a Host Exceptions file if this is a pattern we'll follow more?
Also, consider adding a bit more context. We want this exception message to deflect incidents from reaching our team if a host is using this wrong. 

something like:
"Deferred(Unknown) is not supported in expressions to be evaluated. This is always an error, deferred is only valid when calling Check"

Reply:
Added a constant.